packages/tpetra/core/src/Tpetra_MultiVector_def.hpp

```diff
@@ -1865,6 +1872,7 @@ namespace Tpetra {
           // output buffers, so we have to make a copy of the local
           // sum.
           typename RV::non_const_type lclDots (Kokkos::ViewAllocateWithoutInitializing ("tmp"), numVecs);
+          // DEEP_COPY REVIEW - NOT TESTED
```

Comment:
this is almost certainly copying between the same memory space, but it's not obvious what that space is.
not tested